README.md

```diff
@@ -291,7 +292,77 @@ module.exports = function(defaults) {
   return app.toTree();
 }
 ```
+### Babel config usage
 
+If you want to use the existing babel config from your project instead of the auto-generated one from this addon, then you would need to *opt-in* by passing the config `useBabelConfig: true` as a child property of `ember-cli-babel` in your `ember-cli-build.js` file.
+
+*Note: If you are using this option, then you have to make sure that you are adding all of the required plugins required for Ember to transpile correctly.*
+
+Example usage:
+
+```js
+//ember-cli-build.js
+
+let app = new EmberAddon(defaults, {
+  "ember-cli-babel": {
+    useBabelConfig: true,
+    // ember-cli-babel related options
+  },
+});
+```
+
+```js
+//babel.config.js
+
+const { emberPlugins } = require("ember-cli-babel");
+
+module.exports = function (api) {
+  api.cache(true);
+
+  return {
+    presets: [
+      [
+        require.resolve("@babel/preset-env"),
+        {
+          targets: require("./config/targets"),
+        },
+      ],
+    ],
+    plugins: [
+      // if you want external helpers
+      [
+        require.resolve("@babel/plugin-transform-runtime"),
+        {
+          version: require("@babel/plugin-transform-runtime/package").version,
+          regenerator: false,
+          useESModules: true,
+        },
+      ],
+      [require.resolve("@babel/plugin-proposal-decorators"), { legacy: true }],
+      [require.resolve("@babel/plugin-proposal-class-properties")],
+
+      // this is where all the ember required plugins would reside
+      ...emberPlugins({ ...options }),
+    ],
+  };
+};
+```
+
+#### Ember Plugins
+
+Ember Plugins is a helper function that returns a list of plugins that are required for transpiling Ember correctly. You can import this helper function and add it to your existing `babel.config` file.
+
+**Config options:**
+
+```json
```

Comment:
Lets swap this to `js` file type so it highlights better in the diff

Reply:
👍